### PR TITLE
Added sort option

### PR DESCRIPTION
**Pull request by frepond**
I've added sort option to riak store backend and corresponding tests. Now it supports sort versions of `find_by/6` and `find_all/5`.

### CODE REVIEW

File: src/sumo_store_riak.erl [hunk 1]
```diff
@@ -268,17 +268,34 @@ find_by(DocName, Conditions, undefined, undefined, State) ->
...
+  Query = <<(build_query(TranslatedConditions))/binary>>,
+  case search_keys_by(Conn, Index, Query, SortOpts, Limit, Offset) of
     {ok, {_Total, Keys}} ->
       Results = fetch_docs(DocName, Conn, Bucket, Keys, Opts),
       {ok, Results, State};
     {error, Error} ->
       {error, Error, State}
   end.
 
+-spec count(sumo:schema_name(), state()) -> integer().
```
Comment (gadgetci) [inline src/sumo_store_riak.erl]: According to **Dialyzer**:
> The return type integer() in the specification of count/2 is not a subtype of {'error',_,_} | {'ok',non_neg_integer(),_}, which is the expected return type for the callback of sumo_store behaviour

File: src/sumo_store_riak.erl [hunk 2]
```diff
@@ -268,17 +268,34 @@ find_by(DocName, Conditions, undefined, undefined, State) ->
...
+count(_DocName, _State) ->
```
Comment (gadgetci) [inline src/sumo_store_riak.erl]: According to **Dialyzer**:
> Function count/2 only terminates with explicit exception

File: rebar.config [hunk 3]
```diff
@@ -26,7 +26,7 @@
 
 {deps, [
   {sumo_db, "0.6.4"},
-  {riakc, "2.5.3-beta1"},
+  {riakc, {git, "https://github.com/basho/riak-erlang-client.git", {tag, "2.5.3"}}},
```
Comment (cabol) [inline rebar.config]: We should use the hex version: `{riakc, "2.5.3"}`.

Comment (cabol) [inline rebar.config]: Hi @frepond, I understand, that's annoying, but that's an Inaka's rule, as long as the hex version exist, we should work with that :)

**Pull request merged**